apollo-router/src/plugins/chaos/mod.rs

```diff
@@ -0,0 +1,139 @@
+//! Chaos testing plugin for the Apollo Router.
+//!
+//! This plugin provides chaos testing capabilities to help reproduce bugs that require uncommon conditions.
+//! You probably don't want this in production!
+
+use std::time::Duration;
+
+use futures::Stream;
+use futures::StreamExt;
+use futures::stream;
+use schemars::JsonSchema;
+use serde::Deserialize;
+use serde::Serialize;
+
+pub(crate) mod reload;
+
+// Re-export reload functionality
+pub(crate) use reload::ReloadState;
+
+use crate::router::Event;
+
+/// Configuration for chaos testing, trying to reproduce bugs that require uncommon conditions.
+/// You probably don't want this in production!
+///
+/// ## How Chaos Reloading Works
+///
+/// The chaos system automatically captures and replays the last known schema and configuration
+/// events to force hot reloads even when the underlying content hasn't actually changed. This
+/// is particularly useful for memory leak detection during hot reload scenarios.
+///
+/// ### Schema Reloading (`force_schema_reload`)
+/// When enabled, the router will periodically replay the last schema event with a timestamp
+/// comment injected into the SDL (e.g., `# Chaos reload timestamp: 1234567890`). This ensures
+/// the schema is seen as "different" and triggers a full hot reload, even though the functional
+/// schema content is identical.
+///
+/// ### Configuration Reloading (`force_config_reload`)
+/// When enabled, the router will periodically replay the last configuration event. The
+/// configuration is cloned and re-emitted, which triggers the router's configuration change
+/// detection and reload logic.
+///
+/// ### Automatic Operation
+/// The chaos system requires no manual intervention once configured:
+/// - It automatically captures schema and configuration events as they flow through the system
+/// - Timer-based reloads only occur if there's a "last known" event to replay
+/// - Each chaos timer operates independently - you can enable schema reloading without
+///   configuration reloading, or vice versa
+/// - The system gracefully handles cases where no events are available to replay
```

Comment:
this mostly restates things that were already mentioned, I think it would be clearer if this list was deleted and the idea from the second point (it 'starts working' when config/schema become available) is merged up into the "How Chaos Reloading Works" paragraph.